openml-h2o/src/test/java/com/feedzai/openml/h2o/H2OModelProviderTrainTest.java

```diff
@@ -287,4 +288,186 @@ private void validateParamsForTheAlgorithm(final Map<String, String> params, fin
                 .as("The list parameter validation errors must be null.")
                 .isEqualTo(0);
     }
+
+    /**
+     * Tests H20 parameter validation for NAIVE_BAYES_CLASSIFIER, all parameters correct.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    @Test
+    public final void testValidationOKWhenAllParametersAreValid_NAIVE_BAYES_CLASSIFIER() {
```

Comment:
we are using the same test logic in these tests

they could be moved to a new test class, together with junit parameterized test.
this way we would be defining the parameters of the 7 kind of models, and we would only need to create two tests
- one to validate that the expected parameters of the model work
- and other to validate parameters invalid

Reply:
Great idea! I just finished refactoring this and the new version is way cleaner.